interpreter/llvm/src/tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -364,9 +364,8 @@ def err_thread_unsupported : Error<
 def warn_maybe_falloff_nonvoid_function : Warning<
   "control may reach end of non-void function">,
   InGroup<ReturnType>;
-def warn_falloff_nonvoid_function : Warning<
-  "control reaches end of non-void function">,
-  InGroup<ReturnType>;
+def err_falloff_nonvoid_function : Error<
+  "control reaches end of non-void function">;
```

Comment:
Could you try without changing clang? I.e. neither this change nor the one in interpreter/llvm/src/tools/clang/lib/Sema/AnalysisBasedWarnings.cpp?